docs/en/query_language/select.md

```diff
@@ -136,257 +136,269 @@ Here, the sample of 10% is taken from the second half of data.
 
 ### ARRAY JOIN Clause {#select-array-join-clause}
 
-Allows executing JOIN with an array or nested data structure. The intent is similar to the 'arrayJoin' function, but its functionality is broader.
-
-`ARRAY JOIN` is essentially `INNER JOIN` with an array. Example:
+Allows executing `JOIN` with an array or nested data structure. Allows you to perform `JOIN` both with the external array and with the inner array in the table. The intent is similar to the [arrayJoin](functions/array_functions.md#array_functions-join) function, but its functionality is broader.
 
+``` sql
+SELECT <expr_list>
+FROM <left_subquery>
+[LEFT] ARRAY JOIN <array>
+[WHERE|PREWHERE <expr>]
+...
 ```
-:) CREATE TABLE arrays_test (s String, arr Array(UInt8)) ENGINE = Memory
-
-CREATE TABLE arrays_test
-(
-    s String,
-    arr Array(UInt8)
-) ENGINE = Memory
 
-Ok.
+You can specify only a single `ARRAY JOIN` clause in a query.
 
-0 rows in set. Elapsed: 0.001 sec.
+When running the `ARRAY JOIN`, there is an optimization of the query execution order. Although the `ARRAY JOIN` must be always specified before the `WHERE/PREWHERE` clause, it can be performed as before the `WHERE/PREWHERE` (if its result is needed in this clause), as after completing it (to reduce the volume of calculations). The processing order is controlled by the query optimizer.
 
-:) INSERT INTO arrays_test VALUES ('Hello', [1,2]), ('World', [3,4,5]), ('Goodbye', [])
+Supported types of `ARRAY JOIN` are listed below:
 
-INSERT INTO arrays_test VALUES
+- `ARRAY JOIN` - Executing `JOIN` with an array or nested data structure. Empty arrays are not included in the result.
+- `LEFT ARRAY JOIN` - Unlike `ARRAY JOIN`, when using the `LEFT ARRAY JOIN` the result contains the rows with empty arrays. The value for an empty array is set to 0.
```

Comment:
"is set to 0" -> "is set to default value for array element type (usually 0, empty string or NULL)"